Make kwargs explicit in put, append

jbrockmendel opened the pull request:
Made possible by #29951, progress towards #20903 

@jreback input needed on whether certain not-covered-in-tests cases should be considered possible.  See that both `put` and `append` still contain `**kwargs`.  In all extant tests, kwargs is always empty.  But there are remaining keywords that write_to_group (which both put and append call) accepts that are not in the put/append signatures.

So the question is: is there a reason why put/append would accept different subsets of the writing keywords?  If not, they should both have the same explicit kwargs as write_to_group.

Once this is pinned down, we can get explicit all the way up the stack to NDFrame.to_hdf

jbrockmendel committed:
Make kwargs explicit in put, append

simonjayhawkins commented on pandas/io/pytables.py:
```diff
@@ -1418,17 +1490,16 @@ def _check_if_open(self):
             raise TypeError(f"invalid HDFStore format specified [{format}]")
 
-        return kwargs
+        return format
```
could just return in try block and remove intermediate variable?

simonjayhawkins commented on pandas/core/generic.py:
```diff
@@ -2412,6 +2412,8 @@ def to_hdf(
         complib: Optional[str] = None,
         append: bool_t = False,
         format: Optional[str] = None,
+        min_itemsize=None,
```
could these keywords added here and in #29939 for the public API be keyword only, xref #27544

jbrockmendel replied:
i like this idea, will do in a dedicated PR

simonjayhawkins reviewed (approved):
@jbrockmendel lgtm couple of suggestions, not blockers.

jreback commented:
> Made possible by #29951, progress towards #20903
> 
> @jreback input needed on whether certain not-covered-in-tests cases should be considered possible. See that both `put` and `append` still contain `**kwargs`. In all extant tests, kwargs is always empty. But there are remaining keywords that write_to_group (which both put and append call) accepts that are not in the put/append signatures.
> 
> So the question is: is there a reason why put/append would accept different subsets of the writing keywords? If not, they should both have the same explicit kwargs as write_to_group.
> 
> Once this is pinned down, we can get explicit all the way up the stack to NDFrame.to_hdf

these are likely ok. The reason for extra keywords is mainly on opening of the store itself, where PyTables passes thru keywords (e.g. to specify a different kind of backend: https://github.com/PyTables/PyTables/blob/master/tables/file.py#L219)

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into cln-pytables-kwargs4

jbrockmendel commented:
> these are likely ok. The reason for extra keywords is mainly on opening of the store itself,

To be clear, I'm referring to keywords that we have established are accepted by _write_to_group, i.e. not ones being passed to HDFStore.

jbrockmendel committed:
docstring fix

jreback commented:
> To be clear, I'm referring to keywords that we have established are accepted by _write_to_group, i.e. not ones being passed to HDFStore.

then i think ok to remove

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into cln-pytables-kwargs4

jbrockmendel committed:
types, remove kwargs

jbrockmendel committed:
remove kwargs

jbrockmendel commented:
Updated:

- updated annotations
- removed kwargs
- to do this, had to enforce nan_rep cannot be passed with append=True (double check this is desired)?
- removed no-longer-allowed dropna, fletcher32 kwargs from NDFrame.to_hdf docstring (double check this is desired)?

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into cln-pytables-kwargs4

jbrockmendel committed:
troubleshoot docs build

jbrockmendel commented:
updated+green (ex an unrelated jinja2 problem ill work on in the morning)

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into cln-pytables-kwargs4

jreback commented:
thanks
